### ISSUE TITLE
How to set .graphqlrc programatically ?

### ISSUE THREAD

**Issue by ghost** [kamilkisiela/graphql-config#13]
Hi!
Really enjoy using your software so far! makes things way easier in the complicated graphql/relay world!

I got one question regarding programatically setting grapnel parameters.

The example of `.graphqlrc` shows how to set the `Authorization` header. 
```
{
  "request": {
    "url": "https://example.com/graphql",
    "headers": {
      "Authorization": "xxxxx"
    }
  }
}
```
In my use case, the Auth header is set programatically (JWT token after login). So I'd need the `.graphqlrc` to be set programatically. 

How can I achieve that?

**Comment by schickling** [comment]
Hmm that's a really interesting point. Unfortunately this isn't support as of today but I can totally see this use case.

We're currently redesigning how graphql-config works and will consider this use case as well. Thanks for sharing!

**Comment by kamilkisiela** [comment]
Hi there, it's [The Guild](http://the-guild.dev/) :) We're working on a new version of GraphQL Config. The goal is to make it platform agnostic, flexible enough to use with any tool and framework, reduce the bundle size and a lot more and what's even more important, bring it back to life and keep maintaining.

New changes #162 are already merged into master and `v3.0.0-alpha.13` is released.

We decided to use `cosmiconfig` just like in other libraries we maintain so now you're able to use `yaml`, `json` and even `js` files, [here you can read more](https://github.com/kamilkisiela/graphql-config#possible-config-files).

Using `graphql.config.js` seems like a good fit here.

Because it's still a work in progress, we'd be happy to hear your suggestions and work together on the new version of GraphQL Config.
